### PR TITLE
Feature: Make Visualizer abstract and move Cumulative Return Graph to separate file

### PR DESCRIPTION
**Pull request by travisdula**
I chose to make the the Visualizer class abstract and static, so that we could have all visualization techniques extend it.

### CODE REVIEW

File: olps/visualization/visualizer.py [hunk 1]
```diff
@@ -1,30 +1,15 @@
...
-
-    def visualize(self):
+    @staticmethod
+    @abstractmethod
+    def visualize(outputfile: str, strategies: list[Strategy]):
         """
         Plots the strategies to the output file in a single graph
         """
-        print(f'Attempting to write to {self.outputfile}')
-        for strategy in self.strategies:
```
Comment (adityarathod) [inline olps/visualization/visualizer.py:24]: You used `self.strategies` and `self.outputfile` here, but this is a static method so `self` won't be defined. Either make this a non-static method or use the arguments provided in Line 11.

File: olps/visualization/visualizer.py [hunk 2]
```diff
@@ -1,30 +1,15 @@
...
     """
-    outputfile: str
-    strategies: list[Strategy]
-
-    def __init__(self, outputfile: str, strategies: list[Strategy]):
-        """
-        Initialize the visualizer with already run strategies and output file
-        """
-        self.outputfile = outputfile
-        self.strategies = strategies
```
Comment (adityarathod) [inline olps/visualization/visualizer.py:17]: If you're creating a constructor to store the output file and strategy list, how come it's not used anywhere? `visualize()` is a static method and won't make use of instance variables

File: olps/visualization/visualizer.py (hunk 1 above)
Comment (travisdula) [reply olps/visualization/visualizer.py:24]: These are lines that I deleted

File: olps/visualization/visualizer.py (hunk 2 above)
Comment (travisdula) [reply olps/visualization/visualizer.py:17]: The constructor was deleted, so these are removed lines

**Review by adityarathod — CHANGES_REQUESTED** [review]
I love that you're abstracting away things into general-purpose classes that can be customized and extended.

I have a few questions and changes regarding your usage of static methods which can be detailed in the comments above.

**Review by adityarathod — APPROVED** [review]
Sorry about that! I had a small 🤦 moment. Thanks for pointing out that I was commenting on removed lines.

Looks good!

**Pull request merged**